src/java/org/apache/cassandra/db/SinglePartitionReadCommand.java

```diff
@@ -870,9 +870,9 @@ public void onPartitionClose()
 
     private boolean queriesMulticellType()
     {
-        for (ColumnDefinition column : columnFilter().fetchedColumns())
+        for (ColumnDefinition column : columnFilter().queriedColumns())
         {
-            if (column.type.isMultiCell() || column.type.isCounter())
+            if (column.type.isMultiCell())
```

Comment:
This and the change above are safe because we can't even issue INSERT statements against counter tables? (Might be useful to briefly comment on that somewhere.)